java/src/main/java/ai/rapids/cudf/Rmm.java

```diff
@@ -119,6 +121,27 @@ public static synchronized void initialize(int allocationMode, LogConf logConf,
     initializeInternal(allocationMode, loc.internalId, path, poolSize);
     MemoryCleaner.setDefaultGpu(Cuda.getDevice());
     initialized = true;
+    Rmm.poolingEnabled = isPool || isArena || isAsync;
+    Rmm.poolSize = poolSize;
+  }
+
+  /**
+   * Get the most recently set pool size or -1 if RMM has not been initialized or pooling is
+   * not enabled.
+   */
+  public static synchronized long getPoolSize() {
+    if (isPoolingEnabled()) {
+      return poolSize;
+    } else {
+      return -1;
+    }
+  }
+
+  /**
+   * Return true if rmm is initialized and pooling has been enabled, else false.
+   */
+  public static synchronized boolean isPoolingEnabled() {
+    return initialized && poolingEnabled;
```

Comment:
```suggestion
    return poolingEnabled;
```
`poolingEnabled == true` only after initialization so we don't need to check for `initialized`.